src/components/PopupWithForm.js

```diff
@@ -18,17 +19,40 @@ export default class PopupWithForm extends Popup {
     return formValues;
   }
 
+  _setInputValues(data) {
+    this._inputList.forEach((input) => {
+      input.value = data[input.name] || "";
+    });
+  }
+
   _setEventListeners() {
-    this._popupForm.addEventListener("submit", (event) => {
+    this._popupForm.addEventListener("submit", async (event) => {
       event.preventDefault();
       const data = this._getInputValues();
-      this._handleFormSubmission(data);
-      this.reset();
+      console.log("Form data:", data);
+      this.renderLoading(true);
+      try {
+        await this._handleFormSubmission(data);
```

Comment:
Please, delete `try/catch` from `PopupWithForm`. It doesn't work. You have everything in `index.js`